util/util.go

```diff
@@ -227,3 +227,81 @@ func Match(a interface{}, b interface{}) bool {
 	}
 	return false
 }
+
+//ContainsString checks if we have a string in a string list
+func ContainsString(list []string, value string) bool {
+	if list == nil {
+		return false
+	}
+	for _, item := range list {
+		if value == item {
+			return true
+		}
+	}
+	return false
+}
+
+//ExtendStringList appends a value if it isn't in base list
+func ExtendStringList(target []string, list []string) []string {
+	extended := make([]string, 0, len(target))
+	if target != nil {
+		for _, item := range target {
+			extended = append(extended, item)
+		}
+	}
+
+	if list != nil {
+		for _, item := range list {
+			if !ContainsString(extended, item) {
+				extended = append(extended, item)
+			}
+		}
+	}
+
+	return extended
+}
+
+//ExtendMap extends a map from existing one
+func ExtendMap(target map[string]interface{}, object map[string]interface{}) map[string]interface{} {
```

Comment:
would it make sense to declare the arguments as (target, object map[string]interface{}) ?